dispersy.py

```diff
@@ -1729,25 +1806,28 @@ def store_update_forward(self, possibly_messages, store, update, forward):
                 messages[0].community.statistics.increase_msg_count(u"created", messages[0].meta.name, my_messages)
 
         if forward:
-            return self._forward(messages)
+            forward_result = yield self._forward(messages)
+            returnValue(forward_result)
 
-        return True
+        returnValue(True)
 
     @attach_runtime_statistics(u"Dispersy.{function_name} {1[0].name}")
+    @inlineCallbacks
     def _update(self, messages):
         """
         Call the handle callback of a list of messages of the same type.
         """
         try:
-            messages[0].handle_callback(messages)
-            return True
+            yield messages[0].handle_callback(messages)
+            returnValue(True)
         except (SystemExit, KeyboardInterrupt, GeneratorExit, AssertionError):
```

Comment:
If an exception is raised in a function that is being yielded, it will be reraised by Twisted, so I assume yes?